### PR TITLE
Make configuration md

### PR DESCRIPTION
**Pull request by maho3**
Addresses #71 to add a configuration document explaining all the functionality of every stage of the pipeline

### CODE REVIEW

**Comment by maho3** [comment]
This just needs a once-over to make sure that it's sufficiently helpful to a novice user. Suggestions for additions are welcome, but we can also keep this document in flux as the code develops.

File: CONFIG.md [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+This document provides guidance to configuring an inference pipeline with ltu-ili and includes links and references to the various functionality options. There are three stages to the inference pipeline:
+- [**Data Loading**](#data-loading): Loading various structured data into memory
+- [**Training**](#training): Training neural networks from the loaded data and saving them to file.
+- [**Validation**](#validation): Loading neural networks from file, sampling posteriors on the test set, and computing metrics.
+
+In general, these steps are designed to be independent. For example, if you wanted to try out a different training model, all you would need to do is change the training configuration, and the data loading and validation stages should integrate (nearly) seamlessly with the new model. Each pipeline stage specified below can be instantiated from `json`-like configuration files or from iPython initialization as in [tutorial.ipynb](notebooks/tutorial.ipynb).
+
+## Data Loading
+We have three primary objects for dataloading:
```
Comment (DeaglanBartlett) [inline CONFIG.md:12]: There are four bullet points but you say there are three primary objects. Is one of these deemed to not be primary or should this just be four? If the former, please clarify as it's a little confusing

File: CONFIG.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+xobs_file: 'xobs.npy'  # test data
+thetaobs_file: 'thetaobs.npy'  # test parameters
+
+x_file: 'x.npy'  # filename to save simulated data
+theta_file: 'theta.npy'  # filename to save simulation parameters
+num_simulations: 400  # how many simulations to generate for each training round
+```
+
+You are also welcome also to design your own dataloading objects, so long as they contain the functions: `__len__`, `get_all_data`, and `get_all_parameters`.
```
Comment (DeaglanBartlett) [inline CONFIG.md:55]: Repeated "also"

File: CONFIG.md [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+  class: 'Uniform'
+  args:
+    lower: [0,0,0]
+    upper: [1,1,1]
+
+model:
+  module: 'ili.inference.pydelfi_wrappers'
+  class: 'DelfiWrapper'
+  kwargs:
+    nwalkers: 20 (FIX ME?)
```
Comment (DeaglanBartlett) [inline CONFIG.md:139]: Why is there the entry (FIX ME?)?

**Review by DeaglanBartlett — COMMENTED** [review]
Looks great! Just a couple of potential typos that need fixing before merging

File: CONFIG.md (hunk 1 above)
Comment (maho3) [reply CONFIG.md:12]: :) This was before we merged the jupyter interface. Changed to four.

File: CONFIG.md (hunk 2 above)
Comment (maho3) [reply CONFIG.md:55]: Fixed.

File: CONFIG.md (hunk 3 above)
Comment (maho3) [reply CONFIG.md:139]: I wanted to check that this is the same as the tutorial example, and indeed it is. Removed (FIX ME?)

**Review by DeaglanBartlett — APPROVED** [review]

**Pull request merged**